com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessBuild.cs

```diff
@@ -38,16 +55,35 @@ public void OnPreprocessBuild(BuildReport report)
 
             // If platform is supported all good
             GraphicsDeviceType  unsupportedGraphicDevice = GraphicsDeviceType.Null;
-            if (HDUtils.AreGraphicsAPIsSupported(report.summary.platform, out unsupportedGraphicDevice)
+            bool supported = HDUtils.AreGraphicsAPIsSupported(report.summary.platform, out unsupportedGraphicDevice)
                 && HDUtils.IsSupportedBuildTarget(report.summary.platform)
-                && HDUtils.IsOperatingSystemSupported(SystemInfo.operatingSystem))
-                return;
+                && HDUtils.IsOperatingSystemSupported(SystemInfo.operatingSystem);
 
-            unsupportedGraphicDevice = (unsupportedGraphicDevice == GraphicsDeviceType.Null) ? SystemInfo.graphicsDeviceType : unsupportedGraphicDevice;
-            string msg = "The platform " + report.summary.platform.ToString() + " with the graphic API " +  unsupportedGraphicDevice + " is not supported with High Definition Render Pipeline";
+            if (!supported)
+            {
+                unsupportedGraphicDevice = (unsupportedGraphicDevice == GraphicsDeviceType.Null) ? SystemInfo.graphicsDeviceType : unsupportedGraphicDevice;
+                string msg = "The platform " + report.summary.platform.ToString() + " with the graphic API " + unsupportedGraphicDevice + " is not supported with High Definition Render Pipeline";
 
-            // Throw an exception to stop the build
-            throw new BuildFailedException(msg);
+                // Throw an exception to stop the build
+                throw new BuildFailedException(msg);
+            }
+
+            // Update all quality levels with the right max lod so that meshes can be stripped.
+            // We don't take lod bias into account because it can be overridden per camera.
+            int qualityLevelCount = QualitySettings.names.Length;
+            for (int i = 0; i < qualityLevelCount; ++i)
+            {
+                QualitySettings.SetQualityLevel(i, false);
+                var renderPipeline = QualitySettings.renderPipeline as HDRenderPipelineAsset;
+                if (renderPipeline != null)
+                {
+                    QualitySettings.maximumLODLevel = GetMinimumMaxLoDValue(renderPipeline);
```

Comment:
Don't we need to revert this value somewhere?

Reply:
Thought about this but actually the Quality Settings are hidden to the users anyway and now we state clearly in the doc that they are not supported so using this API would not yield any result (since we override it for each camera afterward).
Also, now when we build a player we override all the various quality level parameters anyway for mesh stripping.

Reply:
I know it is not an issue currently, but with pipeline interoperability, if you switch back to urp after having build HDRP, settings can be a mess. but let's not deal with it for now (but poke jenny about it :) )

Reply:
Yeah, exact same problem as the skybox material. As long as we don't have proper APIs for that we are doomed to have the same issues everywhere.